### PR TITLE
bach inplace materialization (2/2): Support inplace materialization

### PR DESCRIPTION
**Pull request by thijs-obj**
Change: support inplace materialization.

I need this for an upcoming PR, but figured pulling it out makes it a lot easier to review.

This is part two of two PRs:
1. https://github.com/objectiv/objectiv-analytics/pull/328 - move test code
2. https://github.com/objectiv/objectiv-analytics/pull/327 - actual code change <-  **this PR**

### CODE REVIEW

File: bach/bach/dataframe.py [hunk 1]
```diff
@@ -584,31 +584,37 @@ def materialize(self, node_name='manual_materialize', inplace=False, limit: Any
...
-        return self.copy_override(
+
+        df = self.get_instance(
+            engine=self.engine,
             base_node=node,
             index_dtypes=index_dtypes,
-            series_dtypes=series_dtypes,
-            group_by=[None],
-            # materializing resets any sorting as we don't have a way to translate the sorting expressions
-            # to new columns reliably. This needs attention # TODO
```
Comment (thijs-obj) [inline bach/bach/dataframe.py:608]: Removed the TODO as it is documented, and doeesn't really seem like something that we need to change

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -584,31 +584,37 @@ def materialize(self, node_name='manual_materialize', inplace=False, limit: Any
...
         )
 
+        if not inplace:
+            return df
+        self._engine = df.engine
+        self._base_node = df.base_node
+        self._index = df.index
+        self._data = df.data
+        self._group_by = df.group_by
+        self._order_by = df._order_by
```
Comment (jansentom) [inline bach/bach/dataframe.py:615]: Any reason to use the internal one for `order_by` only?

File: bach/tests/functional/bach/test_df_materialize.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 from tests.functional.bach.test_data_and_utils import assert_equals_data, get_bt_with_test_data
 
 
-def test_materialize():
+@pytest.mark.parametrize("inplace", [False, True])
```
Comment (jansentom) [inline bach/tests/functional/bach/test_df_materialize.py:13]: oh, this is nice, didn't know that.

File: bach/bach/dataframe.py (hunk 2 above)
Comment (thijs-obj) [reply bach/bach/dataframe.py:615]: doh. Just inattention. Probably some code completion from pycharm that I used while typing lazily

Comment (thijs-obj) [reply bach/bach/dataframe.py]: changed

**Pull request merged**